src/main/java/com/google/api/gax/grpc/BundlingCallSettings.java

```diff
@@ -1,59 +1,55 @@
 package com.google.api.gax.grpc;
 
-import com.google.common.collect.ImmutableSet;
+import com.google.api.gax.core.RetrySettings;
 
-import io.grpc.ManagedChannel;
 import io.grpc.MethodDescriptor;
+import io.grpc.Status;
 
 import java.io.IOException;
-import java.util.concurrent.ScheduledExecutorService;
+import java.util.Set;
 
 /**
- * A settings class with generic typing that could used to configrue an bundling method
- * or create the bundling callable object, which can be directly operated against an API.
+ * A settings class with generic typing which can be used to configure a bundling method
+ * or create the bundling callable object, which can make API method calls.
  */
 public class BundlingCallSettings<RequestT, ResponseT>
     extends ApiCallSettingsTyped<RequestT, ResponseT> {
-  private BundlingDescriptor<RequestT, ResponseT> bundlingDescriptor;
-  private BundlingSettings bundlingSettings;
+  private final BundlingDescriptor<RequestT, ResponseT> bundlingDescriptor;
+  private final BundlingSettings bundlingSettings;
 
-  ApiCallable<RequestT, ResponseT> create(ServiceApiSettings.Builder serviceSettings)
+  /**
+   * Package-private
+   */
+  ApiCallable<RequestT, ResponseT> create(ServiceApiSettings.Builder serviceSettingsBuilder)
       throws IOException {
-    BundlerFactory<RequestT, ResponseT> bundlerFactory = null;
-    bundlerFactory = new BundlerFactory<>(bundlingDescriptor, bundlingSettings);
+    BundlerFactory<RequestT, ResponseT> bundlerFactory =
+        new BundlerFactory<>(bundlingDescriptor, bundlingSettings);
     BundlingCallable<RequestT, ResponseT> bundlingCallable =
         new BundlingCallable<>(createFutureCallable(), bundlingDescriptor, bundlerFactory);
     ApiCallable<RequestT, ResponseT> callable =
         new ApiCallable<RequestT, ResponseT>(bundlingCallable, this);
-    ManagedChannel channel = serviceSettings.getOrBuildChannel();
-    ScheduledExecutorService executor = serviceSettings.getOrBuildExecutor();
-
-    if (retryableCodes != null) {
-      callable = callable.retryableOn(ImmutableSet.copyOf(retryableCodes));
-    }
-
-    if (retrySettingsBuilder != null) {
-      callable = callable.retrying(retrySettingsBuilder.build(), executor);
-    }
-    callable = callable.bind(channel);
-    return callable;
+    return setupCallable(callable, serviceSettingsBuilder);
```

Comment:
When you create a single createBase() method this problem will go away, but I want to call out that this is the wrong order - retries should be set up "underneath" the bundling, not over it.